### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Drapegnik**
* add User model
* add init users to init-db script
* add user tests
* add session support
* refactor logger setup
* add switch on development and production configs
* add local signin and signup methods with passport.js
* update postman api collection

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,13 +8,14 @@
     "build": "babel . -d build --copy-files --ignore node_modules,postman,build",
     "init-db": "babel-node src/db/init.js",
     "lint": "eslint index.js src/",
-    "test": "mocha --reporter spec src/**/*.test.js --require babel-register"
+    "test": "find ./src -name '*tests.js' | xargs mocha --require babel-register"
```
Comment (Drapegnik) [inline package.json]: i think should be problems with this script for windows users, @tyndria or @asntr check it please.

PS: I don't find how to run mocha with glob pattern, it processed only one test file, mb somebody find solving?

File: src/config/config.json [hunk 2]
```diff
@@ -4,5 +4,21 @@
   "mongodb": {
     "url": "mongodb://localhost/babajka",
     "options": {}
+  },
+  "auth": {
+    "saltRounds": 8
+  },
+  "trustProxy": 0,
+  "session": {
+    "secret": "todo: replace on production",
```
Comment (Drapegnik) [inline src/config/config.json]: @UladBohdan we should discuss how and where we should store this `secret` option

File: src/config/index.js [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+import session from 'express-session';
+import connectMongo from 'connect-mongo';
+import { genSaltSync } from 'bcrypt';
+
 import config from './config.json';
 
+const MongoStore = connectMongo(session);
+
 config.port = process.env.PORT || config.port;
+config.session.secret = genSaltSync(config.auth.saltRounds); // is't ok?
```
Comment (Drapegnik) [inline src/config/index.js]: @UladBohdan, is't ok to set random secret on every run?

File: package.json (hunk 1 above)
Comment (asntr) [inline package.json]: Yep, there're some problems on Windows, I'll try to find a solution.

Comment (asntr) [inline package.json]: Or...We can have a separated folder with tests ;)

Comment (asntr) [inline package.json]: Or I can use cygwin terminal and we won't have any problems.
image

Comment (UladBohdan) [inline package.json]: or you can install Linux

**Comment by UladBohdan** [comment]
all tests are passing for me.

**Comment by UladBohdan** [comment]
Please keep the PR opened
There're some security changes I'm about to push

**Comment by Drapegnik** [comment]
[user model + local auth](https://trello.com/c/cRCcIu4d/8-user-model-local-auth)

**Comment by Drapegnik** [comment]
Updated. @babajka/backend please review

File: src/auth/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      .then(({ id, email, role }) => res.status(200).send({ id, email, role }))
+      .catch(handleError);
+  });
+
+router.post('/register', requireFields('email', 'password'),
+  (req, res, next) => {
+    const handleError = ErrorHandler(res, next);
+    const { password, confirmPassword } = req.body;
+
+    if (password.length < 6) {
```
Comment (asntr) [inline src/auth/index.js]: Why do we validating input on server?

File: src/auth/index.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      return handleError(new ValidationException({ password: 'Password must contain at least 6 characters' }));
+    }
+
+    if (password !== confirmPassword) {
```
Comment (asntr) [inline src/auth/index.js]: This one surely can be detected without requests to server.

File: src/auth/index.js (hunk 4 above)
Comment (asntr) [inline src/auth/index.js]: Hmm...I guess I got it.

File: src/auth/index.js (hunk 5 above)
Comment (Drapegnik) [inline src/auth/index.js]: of course it would be validated on client, but I think it must checked on server side too

Comment (UladBohdan) [inline src/auth/index.js]: no need to send "password confirmation" to the server. We only need it on client for user not to mistype.

File: src/auth/index.js (hunk 4 above)
Comment (UladBohdan) [inline src/auth/index.js]: the validation on the server is a must-have thing

**Comment by Drapegnik** [comment]
Example of `secret.json` file, not required to copy all fields of `config.json`, and nested too
```
{
  "session": {
    "secret": "secret"
  }
}
```

**Pull request merged**